src/state/apis/nft/nftApi.ts

```diff
@@ -245,6 +209,13 @@ export const nftApi = createApi({
 
               data.forEach(item => {
                 const { assetId } = item
+                if (item.collection.isSpam) return
+                if (
+                  nftNameBlacklistRegex.test(
+                    `${item.collection.description} ${item.name} ${item.symbol} ${item.description}`.toLowerCase(),
+                  )
+                )
+                  return
```

Comment:
The crux of this PR, if we upsert portfolio and assets here, we're rugged despite our best efforts everywhere else